admiral/pkg/clusters/virtualservice_routing.go

```diff
@@ -0,0 +1,583 @@
+				"", "", sourceCluster, "remote controller not initialized on this cluster")
+			continue
+		}
+
+		for fqdn, virtualService := range sourceIngressVirtualService {
+			if len(virtualService.Spec.Tls) == 0 {
+				return fmt.Errorf("no TLSRoute found in the ingress virtualservice with host %s", fqdn)
+			}
+			matchesWithNoDestinations := make([]int, 0)
+			for i, tlsRoute := range virtualService.Spec.Tls {
+				tlsMatches := tlsRoute.Match
+				for _, tlsMatch := range tlsMatches {
+					if len(tlsMatch.SniHosts) == 0 {
+						return fmt.Errorf("no SNIHosts found in the ingress virtualservice %s",
+							virtualService.Name)
+					}
+					if len(tlsMatch.SniHosts) > 1 {
+						return fmt.Errorf("more than one SNIHosts found in the ingress virtualservice %s",
+							virtualService.Name)
+					}
+					routeDestinations := destination[defaultFQDN]
+					sniHost := tlsMatch.SniHosts[0]
+					hostWithoutSNIPrefix, err := getFQDNFromSNIHost(sniHost)
+					if err != nil {
+						ctxLogger.Infof(common.CtxLogFormat, "addUpdateVirtualServicesForSourceIngress",
+							virtualService.Name, virtualService.Namespace, sourceCluster, err.Error())
+						continue
+					}
+					if strings.HasPrefix(hostWithoutSNIPrefix, common.BlueGreenRolloutPreviewPrefix) {
+						if destination[previewFQDN] == nil {
+							ctxLogger.Warnf(common.CtxLogFormat, "addUpdateVirtualServicesForSourceIngress",
+								"", "", sourceCluster,
+								"skipped adding preview service, no valid route destinaton found")
+							continue
+						}
+						routeDestinations = destination[previewFQDN]
+					}
+					if strings.HasPrefix(hostWithoutSNIPrefix, common.CanaryRolloutCanaryPrefix) {
+						if destination[canaryFQDN] == nil {
+							ctxLogger.Warnf(common.CtxLogFormat, "addUpdateVirtualServicesForSourceIngress",
+								"", "", sourceCluster,
+								"skipped adding canary service, no valid route destinaton found")
+							continue
+						}
+						routeDestinations = destination[canaryFQDN]
+					}
+					if len(routeDestinations) == 0 {
+						matchesWithNoDestinations = append(matchesWithNoDestinations, i)
+						continue
+					}
+					tlsRoute.Route = routeDestinations
+				}
+
+			}
+
+			// Remove the matches with no destinations
+			for _, indexToBeDeleted := range matchesWithNoDestinations {
+				virtualService.Spec.Tls[indexToBeDeleted] = virtualService.Spec.Tls[len(virtualService.Spec.Tls)-1]
+				virtualService.Spec.Tls = virtualService.Spec.Tls[:len(virtualService.Spec.Tls)-1]
+			}
+
+			existingVS, err := getExistingVS(ctxLogger, ctx, rc, virtualService.Name)
+			if err != nil {
+				ctxLogger.Warn(err.Error())
+			}
+
+			ctxLogger.Infof(common.CtxLogFormat, "addUpdateVirtualServicesForSourceIngress",
+				virtualService.Name, virtualService.Namespace, sourceCluster, "Add/Update ingress virtualservice")
+			err = addUpdateVirtualService(
+				ctxLogger, ctx, virtualService, existingVS, common.GetSyncNamespace(), rc, remoteRegistry)
```

Comment:
checking if this syncNamespace or istio-system

Reply:
@vinay-g , I think syncNamespace will work as well